tests/unit/s2n_tls13_handshake_test.c

```diff
@@ -41,6 +41,59 @@
 
 static int s2n_tls13_conn_copy_server_finished_hash(struct s2n_connection *conn);
 
+static int s2n_test_tls13_handle_secrets(s2n_mode mode, uint8_t version, message_type_t *update_points, size_t update_points_len)
+{
+    for (size_t i = 0; i < S2N_MAX_HANDSHAKE_LENGTH; i++) {
+        struct s2n_connection *conn;
+        notnull_check(conn = s2n_connection_new(mode));
+
+        const struct s2n_ecc_preferences *ecc_pref = NULL;
+        GUARD(s2n_connection_get_ecc_preferences(conn, &ecc_pref));
+        notnull_check(ecc_pref);
+
+        conn->actual_protocol_version = version;
+        conn->secure.cipher_suite = &s2n_tls13_aes_128_gcm_sha256;
+
+        s2n_tls13_connection_keys(client_secrets, conn);
+
+        DEFER_CLEANUP(struct s2n_blob empty_secret, s2n_free);
+        GUARD(s2n_alloc(&empty_secret, client_secrets.size));
+        GUARD(s2n_blob_zero(&empty_secret));
+        eq_check(memcmp(empty_secret.data, client_secrets.extract_secret.data, client_secrets.extract_secret.size), 0);
```

Comment:
Should we be using `EXPECT_BYTEARRAY_EQUAL` instead?

https://github.com/awslabs/s2n/blob/2c0ed13738aee2d8b1c483cad6a1f5f6d4bd59a1/tests/s2n_test.h#L160

Reply:
I'm always torn on how to handle checks in test methods.

The way I'm doing it, the test will exit from where GUARD(s2n_test_tls13_handle_secrets()) is called (telling you which test actually failed) but override the underlying error code + line.

If I instead call S2N_BLOB_EXPECT_EQUAL, the test will immediately exit from that line (which does not tell you which test actually failed, since multiple tests call s2n_test_tls13_handle_secrets) but will preserve the underlying error code + line.

Reply:
Ah i didn't fully realize we were in a helper function here. It's fine to keep it as is, then.